src/storage/mvcc/reader/scanner/forward.rs

```diff
@@ -296,15 +310,23 @@ impl<S: Snapshot, P: ScanPolicy<S>> ForwardScanner<S, P> {
                     // Founded, don't need to seek again.
                     needs_seek = false;
                     break;
+                } else if self.cfg.check_can_be_cached {
+                    self.can_be_cached = false;
                 }
             }
         }
         // If we have not found `${user_key}_${ts}` in a few `next()`, directly `seek()`.
         if needs_seek {
             // `user_key` must have reserved space here, so its clone has reserved space too. So no
             // reallocation happens in `append_ts`.
+            let seek_ts = if self.cfg.check_can_be_cached && self.can_be_cached {
+                // TODO: use "write.seek(ts) + write.prev()" to check newer data maybe faster?
+                TimeStamp::max()
+            } else {
+                self.cfg.ts
+            };
```

Comment:
I guess (not sure) we even don't need to check it here. If we hit `needs_seek`, it means that we must have found some `ts > specified ts`, otherwise the seek bound will not exceed.

Reply:
Good cache! the seeking for check_can_be_cached will not executed on this code

Reply:
Addressed, remove the seeking code